src/routes/safe/components/Transactions/TxsTable/columns.js

```diff
@@ -85,16 +89,27 @@ const getTransactionTableData = (tx: Transaction): TransactionRow => {
     [TX_TABLE_AMOUNT_ID]: getTxAmount(tx),
     [TX_TABLE_STATUS_ID]: tx.status,
     [TX_TABLE_RAW_TX_ID]: tx,
+    [TX_TABLE_RAW_CANCEL_TX_ID]: cancelTx,
   }
 }
 
-export const getTxTableData = (transactions: List<Transaction | IncomingTransaction>): List<TransactionRow> => transactions.map((tx) => {
-  if (tx.type === INCOMING_TX_TYPE) {
-    return getIncomingTxTableData(tx)
-  }
-
-  return getTransactionTableData(tx)
-})
+export const getTxTableData = (
+  transactions: List<Transaction | IncomingTransaction>,
+  cancelTxs: List<Transaction>,
+): List<TransactionRow> => {
+  const cancelTxsByNonce = cancelTxs.reduce((acc, tx) => acc.set(tx.nonce, tx), Map())
+
+  return transactions.map((tx) => {
+    if (tx.type === INCOMING_TX_TYPE) {
+      return getIncomingTxTableData(tx)
+    }
+
+    return getTransactionTableData(
+      tx,
+      Number.isInteger(tx.nonce) ? cancelTxsByNonce.get(tx.nonce) : undefined,
```

Comment:
are we sure that tx.nonce is always a number? `Number.isInteger('0')` would return false.

Maybe worth doing it like that:
`Number.isInteger(+tx.nonce)`

Reply:
Yes, actually... using the `+` operator will lead to false positives. So, lets `parseInt` it before.

```node
> Number.isInteger(Number.parseInt(undefined))
false
> Number.isInteger(Number.parseInt(null))
false
> Number.isInteger(Number.parseInt(false))
false
> Number.isInteger(Number.parseInt(''))
false
> Number.isInteger(Number.parseInt('0'))
true
> Number.isInteger(Number.parseInt(0))
true
>
> Number.isInteger(+undefined)
false
> Number.isInteger(+null)
true
> Number.isInteger(+false)
true
> Number.isInteger(+'')
true
> Number.isInteger(+'0')
true
> Number.isInteger(+0)
true

```